common/src/main/java/com/wynntils/core/functions/FunctionManager.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.functions;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.managers.CoreManager;
+import com.wynntils.functions.TestFunction;
+import com.wynntils.functions.WorldNameFunction;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TextComponent;
+
+/** Manage all built-in {@link Function}s */
+public final class FunctionManager extends CoreManager {
+    private static final List<Function> FUNCTIONS = new ArrayList<>();
+    private static final Set<EnableableFunction> ENABLED_FUNCTIONS = new HashSet<>();
+
+    private static void registerFunction(Function function) {
+        FUNCTIONS.add(function);
+        if (function instanceof EnableableFunction<?> enableableFunction) {
+            enableableFunction.init();
+        }
+    }
+
+    public static List<Function> getFunctions() {
+        return FUNCTIONS;
+    }
+
+    public static boolean enableFunction(Function function) {
+        if (!(function instanceof EnableableFunction<?> enableableFunction)) return true;
+
+        WynntilsMod.getEventBus().register(enableableFunction);
+
+        boolean enableSucceeded = enableableFunction.onEnable();
+
+        if (!enableSucceeded) {
+            WynntilsMod.getEventBus().unregister(enableableFunction);
+        }
+        ENABLED_FUNCTIONS.add(enableableFunction);
+        return enableSucceeded;
+    }
+
+    public static void disableFunction(Function function) {
+        if (!(function instanceof EnableableFunction<?> enableableFunction)) return;
+
+        WynntilsMod.getEventBus().unregister(enableableFunction);
+        enableableFunction.onDisable();
+        ENABLED_FUNCTIONS.remove(enableableFunction);
+    }
+
+    public static boolean isEnabled(Function function) {
+        if (!(function instanceof EnableableFunction<?>)) return true;
+
+        return (ENABLED_FUNCTIONS.contains(function));
+    }
+
+    public static Optional<Function> forName(String functionName) {
+        return FunctionManager.getFunctions().stream()
+                .filter(function -> function.getName().equalsIgnoreCase(functionName))
+                .findFirst();
+    }
+
+    public static Component getSimpleValueString(
+            Function function, String argument, ChatFormatting color, boolean includeName) {
+        MutableComponent header = includeName
+                ? new TextComponent(function.getTranslatedName() + ": ").withStyle(ChatFormatting.WHITE)
+                : new TextComponent("");
+
+        if (function instanceof EnableableFunction<?> enableableFunction && !isEnabled(enableableFunction)) {
```

Comment:
Is this how we want to handle this systematically? Let's say chest count, an enableable function, but it can still produce it's latest active measurement when disabled. Can we someone add this type of functions into the abstraction?

Reply:
What I think we need is separating the "active" part. Is calculation the active part, like current speed or is the "recording"/data collecting the active part? If the later, can't we always spit out the last recorded value?

Reply:
Maybe I'm building an abstraction for a use case that does not exist, but my idea here is that there are potential functions that need to collect data by having a subscription to an event, and cache the last value (or calculate something based on these events, like a moving average, or a sum, etc). Such functions are not "one-shot", like e.g. a function to get your current X and Z position, for which we can instantly query Minecraft.

For this type of functions to be usable, they need to have been "running". I tried to think of a good name, but was not really satisfied with what I could come up with. "ListeningFunction", "BackgroundFunction", "UpdatingFunction", "NonStaticFunction", etc. In the end I crept back to an old OOP practice of just saying what kind of extra method these functions had, namely "enable". I'm still not happy with the name, but it was kind of usable. I do appreciate any suggestions for better name, though! 

An alternative design is to kind of turn this on it's head, assume that all functions are like this, and make the simpler ones a special subclass "StaticFunction". Feels a bit like an anti-OOP-pattern though.

Ok, so given that we assume that this kind of functions need to exist, what should we do if we query them when they are not enabled? Whatever you get back, it's not going to be a valid value. I chose the hard way, of saying "NO!" and clearly marking the result as invalid. But sure, you could certainly return "something" and get a stale value. I'm just worried that we return a misleading value. 

I think basically this boils down to "is it better to say nothing at all unless it is correct, or to try our hardest to give a value even if it's wrong"? I dunno. I prefer the former but maybe I'm a hardliner perfectionist.

Otoh, we are still going to have a situation where a function, once it is enabled, might not have gotten the kind of events it needs, before the first call to `getValue`. So maybe that in itself indicates we should have a more relaxed attitude.

Reply:
I'll try to rewrite it using a more relaxed approach, where the functions are returning cached values if they have.

Reply:
Return cached value but also add a way to see cache age?

Reply:
I was thinking something like that. But it also makes the API more complicated, and I don't know how to present the information to the user. I think some of these details will also have to get straightened out as we start to implement real functions. Right now, it's a lot of guess work.